### PR TITLE
Feat storm lib log (VS-197)

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Add `$lib.log.debug`, `$lib.log.info`, `$lib.log.warning`, and `$lib.log.error` functions. These log to the python logging mechanism under the logger `synapse.storm.log`.
- Functions are gated behind `storm.lib.log.<level>` permissions.
- Functions support a extra arg to allow users to push custom structures into their log values when using structlog.

### CODE REVIEW

File: synapse/lib/stormlib/log.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            This requires the ``storm.lib.log.debug`` permission to use.
+
+        Examples:
+            Log a debug message::
+
+                $lib.log.debug('I am a debug message!')
+
+            Log s debug message with extra information::
+
+                $lib.log.debug('Extra information included here.', extra=$lib.dict(key=$valu))''',
```
Comment (invisig0th) [inline synapse/lib/stormlib/log.py:32]: You should being to use the dict expression syntax :)  ( et al )

File: synapse/lib/stormlib/log.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        if extra and not isinstance(extra, dict):
+            mesg = f'extra provided to log call must be a dictionary compatible type. Got {extra.__class__.__name__} ' \
+                   f'instead.'
+            raise s_exc.BadArg(mesg=mesg, arg='extra')
+        extra = {'synapse': extra}
+        return extra
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _logDebug(self, mesg, extra=None):
+        self.runt.confirm(('storm', 'lib', 'log', 'debug'), gateiden=None)
```
Comment (invisig0th) [inline synapse/lib/stormlib/log.py:140]: dont specify gateiden=None when this is already the default

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/stormlib/log.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                  }
+         },
+        {'name': 'info', 'desc': '''
+            Log a message to the Cortex at the info log level.
+
+            Notes:
+                This requires the ``storm.lib.log.info`` permission to use.
+
+            Examples:
+                Log a info message::
```
Comment (Cisphyx) [inline synapse/lib/stormlib/log.py:50]: 'an info'

File: synapse/lib/stormlib/log.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+                    $lib.log.info('I am a info message!')
+
+                Log a info message with extra information::
```
Comment (Cisphyx) [inline synapse/lib/stormlib/log.py:54]: 'an info'

File: synapse/lib/stormlib/log.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                  }
+         },
+        {'name': 'error', 'desc': '''
+            Log a message to the Cortex at the error log level.
+
+            Notes:
+                This requires the ``storm.lib.log.error`` permission to use.
+
+            Examples:
+                Log a error message::
```
Comment (Cisphyx) [inline synapse/lib/stormlib/log.py:98]: 'an error'

File: synapse/lib/stormlib/log.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/stormlib/log.py:140]: fixed

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**